pr_agent/git_providers/git_provider.py

```diff
@@ -75,20 +75,33 @@ def get_pr_description(self, *, full: bool = True) -> str:
     def get_user_description(self) -> str:
         description = (self.get_pr_description_full() or "").strip()
         description_lowercase = description.lower()
+
         # if the existing description wasn't generated by the pr-agent, just return it as-is
         if not self._is_generated_by_pr_agent(description_lowercase):
             return description
-        # if the existing description was generated by the pr-agent, but it doesn't contain the user description,
+
+        # if the existing description was generated by the pr-agent, but it doesn't contain a user description,
         # return nothing (empty string) because it means there is no user description
         user_description_header = "## user description"
         if user_description_header not in description_lowercase:
             return ""
+
         # otherwise, extract the original user description from the existing pr-agent description and return it
-        user_description_start_position = description_lowercase.find(user_description_header) + len(user_description_header)
-        return description[user_description_start_position:].split("\n", 1)[-1].strip()
+        # user_description_start_position = description_lowercase.find(user_description_header) + len(user_description_header)
+        # return description[user_description_start_position:].split("\n", 1)[-1].strip()
+
+        # the 'user description' is in the beginning. extract and return it
+        original_user_description = description.split("___")[0].strip()
```

Comment:
If the user description itself contains `___` this will only grab part of it.  
Maybe find the position of `## user description` and the first position of one of the possible headers and grab the substring between them?

Something like this maybe?
```python
start_position = description_lowercase.find(user_description_header) + len(user_description_header)
end_position = min(description_lowercase.find(header) for header in possible_headers if header != user_description_header)
original_user_description = description[start_position:end_position]
```